### ISSUE TITLE
[Widgets are swapped] : Using this with different types of widgets produces weird result.

### ISSUE THREAD

**Issue by bimsina** [letsar/local_hero#3]
Let me know if I am missing something, but when I try to move between two different types of widgets, the widgets to be rendered is swapped in the next transition. 

<img src = "https://user-images.githubusercontent.com/29589003/115552744-5c453680-a2cc-11eb-9e49-cd1b135a3a35.gif" width ="250"></img>

<details>
  <summary>Top Widget</summary>

```
Container(
          key: ValueKey(tile),
          child: LocalHero(
            tag: tile.text,
            child: GestureDetector(
              onTap: () {
                setState(() {
                  colTiles.add(tile);
                  rowTiles.remove(tile);
                });
              },
              child: Container(
                color: tile.color,
                height: 60,
                width: 60,
                child: Padding(
                  padding: const EdgeInsets.all(16),
                  child: CircleAvatar(
                    backgroundColor: Colors.white70,
                    foregroundColor: Colors.black54,
                    child: Text(tile.text),
                  ),
                ),
              ),
            ),
          ),
        )
```
</details>

<details>
  <summary>Bottom Widget</summary>

```
Container(
                  key: ValueKey(tile),
                  child: LocalHero(
                    tag: tile.text,
                    child: GestureDetector(
                      onTap: () {
                        setState(() {
                          rowTiles.add(tile);
                          colTiles.remove(tile);
                        });
                      },
                      child: Container(
                        width: 80,
                        height: 80,
                        color: tile.color,
                      ),
                    ),
                  ),
                )
```
</details>

So basically the top widget has text and the bottom doesn't.

The issue is TopWidget is rendered in the bottom column in the next transition if the item was previously a part of the top row.

[Here](https://pastebin.com/raw/9Y6hTNaz) is the code if you want to test it out yourself.

**Comment by arxdeus** [comment]
how's u fixed that?
